lec02-basic-r.Rmd

```diff
@@ -511,245 +535,347 @@ the hierarchy of how these data types are coerced?
+surveys <- readr::read_csv('portal_data.csv')
+```
+
+```{r, echo=FALSE}
+surveys <- readr::read_csv('data/portal_data.csv')
+```
+
+This statement produces some output regarding which data type it found in each 
+column. If we want to check this in more detail, we can print the variable's 
+value: `surveys`.
 
 ```{r}
-animals <- c("mouse", "rat", "dog", "cat")
-animals[animals == "cat" | animals == "rat"] # returns both rat and cat
-animals %in% c("rat", "cat", "dog", "duck", "goat")
-animals[animals %in% c("rat", "cat", "dog", "duck", "goat")]
+surveys
 ```
 
-#### Challenge
+In the online html-version of this lecture, you only see the first few rows of
+the data frame. Running the code chunk above in the R Notebook would display a
+nice tabular view of the data, which also includes pagination when there are
+many rows and we can click the green arrow to view all the columns. Technically,
+this object actually a `tibble` rather than a data frame, as indicated in the
+output. The reason for this is that `read_csv` automatically converts the data
+into to a `tibble` when loading it. Since a `tibble` is just a data frame with
+some convenient extra functionality, we will use these words interchangeably
+from now on.
 
-* Can you figure out why `"four" > "five"` returns `TRUE`?
+If we just want to glance at how the data frame looks, it is sufficient to display only the
+top (the first 6 lines) using the function `head()`:
 
 ```{r}
-# Answers
-# * When using ">" or "<" on strings, R compares their alphabetical order. Here
-#   "four" comes after "five", and therefore is "greater than" it.
+head(surveys)
 ```
 
-## Missing data
+## What are data frames?
 
-As R was designed to analyze datasets, it includes the concept of missing data
-(which is uncommon in other programming languages). Missing data are represented
-in vectors as `NA`.
+Data frames are the _de facto_ data structure for most tabular data, and what we
+use for statistics and plotting. A data frame can be created by hand, but most
+commonly they are generated by the function `read_csv()`. In other words, when
+importing spreadsheets from your hard drive (or the web).
 
-When doing operations on numbers, most functions will return `NA` if the data
-you are working with include missing values. This feature
-makes it harder to overlook the cases where you are dealing with missing data.
-You can add the argument `na.rm=TRUE` to calculate the result while ignoring
-the missing values.
+A data frame is the representation of data in the format of a table where the
+columns are vectors that all have the same length. Because the columns are
+vectors, they all contain the same type of data as we discussed before the break
+(e.g., characters, integers, factors). We can see this when inspecting the
+structure of a data frame with the function `str()`:
 
 ```{r}
-heights <- c(2, 4, 4, NA, 6)
-mean(heights)
-max(heights)
-mean(heights, na.rm = TRUE)
-max(heights, na.rm = TRUE)
+str(surveys)
```

Comment:
Worth mentioning that they see some of this info without `str` for tibbles? Or unnecessary maybe.

Reply:
Perhaps a quick verbal mention - I don't know if it needs to be a separate point here, however. I feel that emphasizing it might confuse students into thinking that printing out the entirety of data frames in general (ie not just tibbles) will display that information.